docs/user/01-20-istio-custom-resource.md

```diff
@@ -0,0 +1,43 @@
+# Istio custom resource
+
+The `istios.operator.kyma-project.io` CustomResourceDefinition (CRD) describes the kind and the format of data that Istio Controller uses to configure, update, and manage the Istio installation. Applying the CR triggers the installation of Istio, and deleting it triggers the uninstallation of Istio. To get the up-to-date CRD in the `yaml` format, run the following command:
+
+```shell
+kubectl get crd istios.operator.kyma-project.io -o yaml
+```
+
+## Specification
+
+This table lists all the possible parameters of the given resource together with their descriptions:
+
+
+**Spec:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **components**  | object |  |
+| **components.cni**  | object | Defines component configuration for Istio CNI DaemonSet. |
+| **components.cni.k8s.affinity**  | object | Affinity is a group of affinity scheduling rules. To learn more, read about affininty in the [Istio documentation](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Affinity).|
+| **components.cni.k8s.resources**  | object | Defines [Kubernetes resources configuration](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/). For more information, read about Resources in the [Istio documentation](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Resources ). |
+| **components.ingressGateway**  | object | Defines component configurations for Istio Ingress Gateway. |
+| **components.ingressGateway.k8s.hpaSpec**  | object | Defines configuration for HorizontalPodAutoscaler. |
+| **components.ingressGateway.k8s.hpaSpec.maxReplicas**  | integer |  |
+| **components.ingressGateway.k8s.hpaSpec.minReplicas**  | integer |  |
+| **components.ingressGateway.k8s.resources**  | object | Defines [Kubernetes resources configuration](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/). To learn more, read the [Istio documentation](https://istio.io/latest/docs/reference/config/istio.operator.v1). |
+| **components.ingressGateway.k8s.strategy**  | object | Defines the rolling update strategy. To learn more, read about DeploymentStrategy in the [Istio documnetation](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#DeploymentStrategy). |
+| **components.pilot**  | object | Defines component configuration for Istiod. |
+| **components.pilot.k8s.hpaSpec**  | object | Defines configuration for HorizontalPodAutoscaler. |
+| **components.pilot.k8s.hpaSpec.maxReplicas**  | integer |  |
```

Comment:
We can use the description from https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#horizontalpodautoscalerspec-v1-autoscaling .